src/plugins/intel_cpu/src/emitters/snippets/x64/jit_parallel_loop_emitters.cpp

```diff
@@ -0,0 +1,417 @@
+                              in.size());
+    OV_CPU_JIT_EMITTER_ASSERT(is_incremented.size() == io_num,
+                              "Invalid is_incremented size: expected ",
+                              io_num,
+                              " got ",
+                              is_incremented.size());
+    OV_CPU_JIT_EMITTER_ASSERT(loop_end_label != nullptr && loop_begin_label != nullptr, "has not inited labels!");
+    OV_CPU_JIT_EMITTER_ASSERT(!snippets::utils::is_dynamic_value(wa_increment) || evaluate_once,
+                              "loop increment might be dynamic only if loop evaluates once!");
+}
+
+void jit_parallel_loop_end_emitter::emit_code_impl(const std::vector<size_t>& in,
+                                                   const std::vector<size_t>& out,
+                                                   const std::vector<size_t>& pool_vec_idxs,
+                                                   const std::vector<size_t>& pool_gpr_idxs) const {
+    validate_arguments(in, out);
+    jit_emitter::emit_code_impl(in, out, pool_vec_idxs, pool_gpr_idxs);
+}
+
+void jit_parallel_loop_end_emitter::emit_impl(const std::vector<size_t>& in,
+                                              [[maybe_unused]] const std::vector<size_t>& out) const {
+    if (!evaluate_once) {
+        Reg64 reg_increments;
+        auto add_increments = [&]() {
+            for (size_t idx = 0; idx < mem_ptr_regs_idxs.size(); idx++) {
+                const auto& increment = loop_args.m_ptr_increments[idx];
+                if (is_incremented[idx] && increment != 0) {
+                    if (ov::snippets::utils::is_dynamic_value(increment)) {
+                        OV_CPU_JIT_EMITTER_ASSERT(is_dynamic, "Loop argument structure cannot be pushed to aux GPR");
+                        h->add(Reg64(static_cast<int>(mem_ptr_regs_idxs[idx])),
+                               h->ptr[reg_increments + idx * sizeof(int64_t)]);
+                    } else {
+                        h->add(Reg64(static_cast<int>(mem_ptr_regs_idxs[idx])), increment);
+                    }
+                }
+            }
+        };
+
+        if (is_dynamic) {
+            jit_aux_gpr_holder gpr_holder(h, aux_gpr_idxs, in);
+            reg_increments = gpr_holder.get_reg();
+            h->mov(reg_increments, wa_increment);
+            h->mov(reg_increments, h->ptr[abi_param1 + GET_OFF(loop_args)]);
+            h->mov(reg_increments, h->ptr[reg_increments + loop_id_offset + GET_OFF_LOOP_ARGS(m_ptr_increments)]);
+            add_increments();
+        } else {
+            add_increments();
+        }
+
+        auto reg_work_amount = Reg64(in.back());
+        h->sub(reg_work_amount, wa_increment);
+        h->cmp(reg_work_amount, wa_increment);
+        h->jge(*loop_begin_label, CodeGenerator::T_NEAR);
+    }
+    m_parallel_section_reg_spiller->postamble();
+    // Note: parallel region ends here:
+    h->ret();
+    h->L(*loop_end_label);
```

Comment:
Could you please add comment with brief explanation: where is applying of `finalization offsets` and why are they there?
To recall it in the future